iOS/UI/Browse/BrowseViewController.swift

```diff
@@ -254,14 +311,59 @@ extension BrowseViewController {
 extension BrowseViewController {
 
     enum Section: Int {
-//        case pinned
+        case pinned
         case updates
         case installed
         case external
     }
 
+    // Ability to edit tableview for a diffable data source.
+    // Changing data in a diffable data source requires its seperate tableview override which can't be done with the view's tableview delegate.
+    class SourceCellDataSource: UITableViewDiffableDataSource<Section, SourceInfo2> {
+        // Let the rows in the Pinned section be reordered (used for reordering sources)
+        override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
+            if #available(iOS 15.0, *) {
+                return sectionIdentifier(for: indexPath.section) == .pinned
+            } else {
+                let section = indexPath.section
+                guard section >= 0 else { return false }
+                let sections = self.snapshot().sectionIdentifiers
+                return ((sections.count > section ? sections[section]: Section.installed) == Section.pinned)
+            }
+        }
+
+        // Move a selected source row from pinned section to a destination index.
+        override func tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
+            guard let sourceItem = itemIdentifier(for: sourceIndexPath) else { return }
+            guard sourceIndexPath != destinationIndexPath else { return }
+
+            let destinationItem = itemIdentifier(for: destinationIndexPath)
+
+            var snapshot = self.snapshot()
+
+            if let destinationItem = destinationItem {
+                if let sourceIndex = snapshot.indexOfItem(sourceItem),
+                   let destinationIndex = snapshot.indexOfItem(destinationItem) {
+
+                    snapshot.deleteItems([sourceItem])
+
+                    if destinationIndex > sourceIndex {
+                        snapshot.insertItems([sourceItem], afterItem: destinationItem)
+                    } else {
+                        snapshot.insertItems([sourceItem], beforeItem: destinationItem)
+                    }
+                }
+            }
+            // Save the order and notify the observer to reload table.
+            apply(snapshot, animatingDifferences: false)
+            NotificationCenter.default.post(name: Notification.Name("updatePinnedList"), object: nil)
```

Comment:
I don't think you should use notifications here, they're generally not a good option. probably use a weak var for the parent as a delegate since SourceCellDataSource is only used in this view controller.